ManipulaPy/potential_field.py

```
@@ -95,81 +158,101 @@ class CollisionChecker:
+                # Fallback for legacy mesh attribute
+                elif hasattr(geom, "mesh") and geom.mesh is not None:
+                    mesh = geom.mesh
+                    if hasattr(mesh, "vertices") and mesh.vertices is not None:
+                        vertices = np.array(mesh.vertices)
+                        if len(vertices) >= 4:
                             try:
                                 convex_hull = ConvexHull(vertices)
                                 convex_hulls[link.name] = convex_hull
                             except Exception:
-                                # Skip if convex hull fails (degenerate geometry)
                                 pass
```

Pylint (4.0.5)

[refactor] Too many nested blocks (6/5)

(R1702)